src/Database.js

```diff
@@ -38,21 +38,31 @@ class Database {
     });
   }
 
-  run(sql, ...params) {
-    return new this.Promise((resolve, reject) => {
-      this.driver.run(sql, params || [], function runExecResult(err) {
+  run(sql) {
+    const params = (arguments.length === 2 && typeof arguments[1] === 'object'
```

Comment:
I pushed [a commit](https://github.com/kriasoft/node-sqlite/pull/10/commits/bf0df1b58f6631d378e170d757f3429fa8418adb) to implement this refactoring and [a subsequent simplification](https://github.com/kriasoft/node-sqlite/pull/10/commits/9296e97e12acffef1ce166affe1a1fedc3511a32).